File: Sample codes/Offline Identity Verification/PHP/offline_identity_verification.php
<?php
$url="https://api.shuftipro.com/";
$client_id  = "Your client id provided by Shufti Pro";
$secret_key = 'YOUR_SECRET_KEY'; //Replace with your secret key provided by the Shufti Pro.


$verification_services = array(
	"document_type"        => "passport",
	"document_id_no"       => "6980XYZ4821XYZ",
	"first_name"           => "John",
	"last_name"            => "Doe",
	"dob"                  => "1900-01-01",
	"background_checks"    => "0"
);
//JSON encode the services array
$verification_services = json_encode($verification_services);

$sample_face_image = "https://raw.githubusercontent.com/shuftipro/integration-guide/master/assets/realFace.jpg";
$sample_id_image = "https://raw.githubusercontent.com/shuftipro/integration-guide/master/assets/realId.jpg";


$verification_data = array(
	"face_image"             => base64_encode(file_get_contents($sample_face_image)),
	"document_front_image"   => base64_encode(file_get_contents($sample_id_image))
);
//JSON encode the services array
$verification_data = json_encode($verification_data);


$post_data = array(
	"client_id"             => $client_id,
	"reference"             => "ref-" . rand(1000,100000),
	"email"                 => "customer@gmail.com",
	"phone_number"          => "+440000000000",
	"country"               => "us",
	"lang"                  => "en",
	"callback_url"          => "https://www.yourdomain.com",
	"redirect_url"          => "https://www.yourdomain.com",
	"verification_services" => $verification_services,
	"verification_data"     => $verification_data
);

ksort($post_data);//Sort the all request parameter.
$raw_data = implode("", $post_data) . $secret_key; //Replace with your secret key provided by the Shuftipro;

$signature              = hash("sha256", $raw_data);
$post_data["signature"] = $signature;

$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, $url);
curl_setopt($ch, CURLOPT_POST, 1);
curl_setopt($ch, CURLOPT_POSTFIELDS, $post_data);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
$response = curl_exec($ch);
curl_close($ch);

$response = json_decode($response);
$my_signature = hash("SHA256", $response->status_code . $response->message . $response->reference . $secret_key);


if($my_signature == $response->signature){
		# Response is valid.
		echo $response->message;
}
else{
	echo "Response signature is invalid";
}
?>

File: Sample codes/Offline Identity Verification/Python/offline_identity_verification.py
#!/usr/bin/python
import hashlib
import requests
import json
import collections
import random
import base64
import requests
from decimal import Decimal

url = 'https://api.shuftipro.com/'
CLIENT_ID = 'Your client id provided by Shuftipro'
SECRET_KEY = 'YOUR_SECRET_KEY'



verification_services ={}
verification_services["document_type"]        = "passport"
verification_services["document_id_no"]       = "6980XYZ4521XYZ"
verification_services["first_name"]           = "John"
verification_services["last_name"]            = "Doe"
verification_services["dob"]                  = "1900-01-01"
verification_services["background_checks"]    = "0"

#json encode vericiation services
json_verification_services = json.dumps(verification_services, ensure_ascii=False)

verification_data ={}

#Sample Face Image
sample_face_image = "https://raw.githubusercontent.com/shuftipro/integration-guide/master/assets/realFace.jpg"
sample_id_image = "https://raw.githubusercontent.com/shuftipro/integration-guide/master/assets/realId.jpg"

verification_data["face_image"]             = base64.b64encode(requests.get(sample_face_image).content)
verification_data["document_front_image"]   = base64.b64encode(requests.get(sample_id_image).content)


#json encode vericiation data
json_verification_data = json.dumps(verification_data, ensure_ascii=False)

post_data = {
"client_id"             : CLIENT_ID,
"reference"             : "ref-" + str(random.randint(1000,10000)),
"email"                 : "customer@email.com",
"phone_number"          : "+440000000000",
"country"               : "it",
"lang"                  : "de",
"callback_url"          : "https://www.yourdomain.com",
"redirect_url"          : "https://www.yourdomain.com",
"verification_services" : json_verification_services,
"verification_data"     : json_verification_data
}

post_data = collections.OrderedDict(sorted(post_data.items())) #sort the dictionary
raw_data = "".join(post_data.values()) + SECRET_KEY #get values from dictionary and append secret key

hash_object = hashlib.sha256(raw_data) #calculating sha 256 hash
signature = hash_object.hexdigest()

post_data['signature'] = signature #append signature to data dictionary
response = requests.post(url, post_data).json() #send POST request to API

#Validate response signature
my_signature = hashlib.sha256(response["status_code"] + response["message"]  +response["reference"] + SECRET_KEY).hexdigest()

if my_signature == response["signature"]:
    # Response is valid.
    print response["message"];
else:
    print "Response signature is invalid";
